src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -472,6 +466,33 @@ public static IEnumerable GetEnumerable(object obj)
                 objectType = obj.GetType();
             }
 
+            return objectType;
+        }
+
+        /// <summary>
+        /// Returns True if the language considers obj to be IEnumerable
+        /// </summary>
+        /// <param name="obj">
+        /// IEnumerable or IEnumerable-like object
+        /// </param>
+        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
+        public static bool IsObjectEnumerable(object obj)
+        {
+            return IsTypeEnumerable(GetBaseObjectType(obj));
+        }
+
+
+        /// <summary>
+        /// Retrieves the IEnumerable of obj or null if the language does not consider obj to be IEnumerable
+        /// </summary>
+        /// <param name="obj">
+        /// IEnumerable or IEnumerable-like object
+        /// </param>
+        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
+        public static IEnumerable GetEnumerable(object obj)
+        {
+            Type objectType = GetBaseObjectType(obj);
+            if (objectType == null) { return null; }
             GetEnumerableDelegate getEnumerable = GetOrCalculateEnumerable(objectType);
             return getEnumerable(obj);
```

Comment:
This is wrong. Before the change, the `obj` passed in `getEnumerable` is the base object if the original one is PSObject, but now it will be the PSObject in that case.

Reply:
Oops, you're right.  Fixing now.